index.bs

```diff
@@ -5294,38 +5294,40 @@ partial interface MLGraphBuilder {
     // of the input values itself, in order to increase the numerical stability of
     // the result.
     // [1]: https://cs231n.github.io/linear-classify/#softmax
-    const max_x = builder.reduceMax(x, { axes: [1], keepDimensions: true });
+    const max_x = builder.reduceMax(x, { axes: [axis], keepDimensions: true });
     const exp_x = builder.exp(builder.sub(x, max_x));
-    return builder.div(exp_x, builder.reduceSum(exp_x, { axes: [1], keepDimensions: true }));
+    return builder.div(exp_x, builder.reduceSum(exp_x, { axes: [axis], keepDimensions: true }));
   </pre>
 </details>
 </div>
 
-#### {{MLGraphBuilder/softmax(input)}} #### {#api-mlgraphbuilder-softmax-input}
+#### {{MLGraphBuilder/softmax(input, axis)}} #### {#api-mlgraphbuilder-softmax-input-axis}
 <div>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input 2-D tensor.
+        - *axis*: an {{unsigned long}} scalar. The dimension the softmax will be performed on.
 
     **Returns:**
         - an {{MLOperand}}. The output 2-D tensor that contains the softmax results, of the same shape as *input*.
 </div>
 
 <details open algorithm>
   <summary>
-    The <dfn method for=MLGraphBuilder>softmax(|input|)</dfn> method steps are:
+    The <dfn method for=MLGraphBuilder>softmax(|input|, |axis|)</dfn> method steps are:
   </summary>
     1. If [=MLGraphBuilder/validating operand=] with [=this=] and |input| returns false, then [=exception/throw=] a {{TypeError}}.
     1. If |input|'s [=MLOperand/rank=] is not 2, then [=exception/throw=] a {{TypeError}}.
```

Comment:
```suggestion
    1. If |input|'s [=MLOperand/rank=] is less than 1, then [=exception/throw=] a {{TypeError}}.
```

Reply:
Don't we need `> 0` in there, to preclude scalars?

Reply:
> Don't we need `> 0` in there, to preclude scalars?

The line below it says:

`1. If |axis| is greater than or equal to |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.`

Scalars have a rank == 0. So passing axis == 0 will cause a `TypeError` since 0 >= 0.
We can add > 0 for redundancy 🤷.

Reply:
Oh, duh, thanks for reading the condition for me. Unnecessary line removed in 1877f2eb13c35a375c7ca86b112dc27ace7d9a4e